jumble/src/views/CharmDetailView.tsx

```diff
@@ -273,34 +274,37 @@ function useCodeEditor(
       } else {
         setWorkingSrc(extractUserCode(iframeRecipe.src ?? "") ?? "");
       }
+      setWorkingSpec(iframeRecipe.spec);
     }
   }, [iframeRecipe, charm, showFullCode]);
 
   const hasUnsavedChanges = showFullCode
-    ? workingSrc !== iframeRecipe?.src
-    : injectUserCode(workingSrc ?? "") !== iframeRecipe?.src;
+    ? workingSrc !== iframeRecipe?.src || workingSpec !== iframeRecipe?.spec
+    : injectUserCode(workingSrc ?? "") !== iframeRecipe?.src || workingSpec !== iframeRecipe?.spec;
```

Comment:
This is a bit ugly, can we refactor to a clearer format? Pattern matching, variables.